### ISSUE TITLE
Can this be used to upload files to Dropbox? How?

### ISSUE THREAD

**Issue by vladejs** [23/resumable.js#374]
Hi,

I am having problems to upload big files to dropbox through my app. All files under 150 mb get uploaded fine using the simple XMLHttpRequest interface, but with bigger files I get a `connection reset` error. I am trying this library to see if it solves my problem, but I am receiving the following error:
```
content.dropboxapi.com/2/files/upload?resumableChunkNumber=1&resumableChunk…
ename=toilet.jpg&resumableRelativePath=toilet.jpg&resumableTotalChunks=1:1 
GET https://content.dropboxapi.com/2/files/upload?resumableChunkNumber=1&resuma…
ilename=toilet.jpg&resumableRelativePath=toilet.jpg&resumableTotalChunks=1 400 (Bad Request)
```

This is the code I use to upload one file:
```
const r = new Resumable({
    target: 'https://content.dropboxapi.com/2/files/upload',
    headers: {
      'Authorization': `Bearer ${dropboxToken}`,
      'Content-Type': 'application/octet-stream',
      'Dropbox-API-Arg': JSON.stringify({
        path:       '/' + file.name,
        mode:       'add',
        autorename: true,
        mute:       false
      })
    },
    query: {}
  });

  r.on('fileAdded', function(file, event){
    r.upload();
  });

  r.on('fileSuccess', function(file, message){
    resolve(file);
  });

  r.on('fileError', function(file, message){
    reject(message);
  });

  r.on('fileProgress', function(file){
    const progress = Math.floor(file.progress() * 100);
    console.log('on file progress', progress);
  });

  r.addFile(file);
```

Anyone have a working example with Dropbox?

**Comment by cpnielsen** [comment]
Hi! I can't seem to find the documentation for your specific endpoint, but I suggest you use one of the provided (third-party) javascript libraries for Dropbox found here: https://www.dropbox.com/developers-v1/core/sdks/other.

We cannot offer support for third-party APIs, only for the library itself.

**Comment by vladejs** [comment]
@cpnielsen , You are pointing me on the wrong direction. Dropbox's SDK v1 **is deprecated**. Here are all the endpoints documented for the SDK v2: 

https://www.dropbox.com/developers/documentation/http/documentation

As stated in my first comment, I can successfully upload files with the ol' simple `XMLHttpRequests`, but I got in trouble with big files. I think I should use session or one of the endpoints that accept multiple chunks. 

Again, the problem is I don't know how to use resumable.js with those endpoints

**Comment by cpnielsen** [comment]
@vladejs There is an official, updated (although work-in-progress) SDK from Dropbox for v2 here: https://www.dropbox.com/developers/documentation/javascript#overview

The documentation for Dropbox also specifically says not to use `/files/upload` for files over 150 MB (see https://www.dropbox.com/developers/documentation/http/documentation#files-upload), but to use `/files/upload_session/start` instead.

A quick read of the documentation indicates that their chunk-based upload is akin to Resumable.js, but simpler and you would have to modify what data is sent with each chunk (session_id + offset calculation). That is left as an exercise for the reader.

**Comment by vladejs** [comment]
Problem is, that being the SDK v2 a work in progress, they don't have examples to do it with their SDK, I found a SO question which gives an example, but it doesn't work for me. Also, it doesn't show a way to get the upload progress.

That's why I'am trying to use Resumable.js

**Comment by vladejs** [comment]
Could you give me some help in here @cpnielsen ?

**Comment by cpnielsen** [comment]
I am afraid resumable.js is not the right tool for this job without some rewrites - by the looks of it, Dropbox seems to expect sequential uploads (Resumable.js performs uploads as fast as possible and does not care about order) and only one at a time. Resumable.js can help you chunk the files, but the upload mechanism and chunk-checking mechanism would have to be completely rewritten.

The Dropbox SDK does have a javascript upload example (in the example folder) showcasing how to use it, and it looks fairly straightforward. I am sorry we cannot help you any further.

**Comment by vladejs** [comment]
Thank you for your reply. 

Last night I tried to use the Dropbox API manually but it seems that my chunking algorithm doesn't worked as expected (I modified and used [this](https://gist.github.com/alediaferia/cfb3a7503039f9278381)). 

The upload example on dropbox repo is for simple uploads (files lower than 150mb), and they don't have a clear explanation in how to upload in chunks using the sdk. I created [this issue](https://github.com/dropbox/dropbox-sdk-js/issues/120). 

I'll keep looking.
